### PR TITLE
Wheater App

### PR DESCRIPTION
**Pull request by NataJenkins**
Wheater app created using javascript where data is fetched through an api

### CODE REVIEW

File: src/app/index.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+const storage = new Store();
+const { city, countryCode } = storage.getLocationData();
+const weather = new Weather(city, countryCode);
+
+// User Interface
+const ui = new UI();
+
+require('./index.css');
+
+async function fetchWeather() {
```
Comment (andresporras3423) [inline src/app/index.js:14]: use arrow functions whenever possible

**Review by andresporras3423 — CHANGES_REQUESTED** [review]
### Status: Changes Required :recycle:
**Hello  Nat Macias Ortiz** 
Good job with the improvements for this project :muscle:  :muscle:
It meets most of the requirements :clap: :clap: :clap:  
There are just a few details to improve before you can go ahead:

- [x] In your package.json file, add the next:
```
"scripts": {
    "dev": "webpack --mode development",
    "build": "webpack --mode production",
    "start": "webpack-dev-server --mode development --open" // optional
  }
```
- [x]  changes the page appearance based on the received data
- [ ]  It should have a toggle control to determine if the result will be shown in Celsius or Fahrenheit.
- [x] In your README file, explain how to run this project.
- [ ] Use arrow functions whenever possible

#### OPTIONAL

- Please, next time don't close a pull request before approval
 
Please request another review once you are ready! :+1: 
 
Happy coding :relaxed:
 
Oscar Andrés Russi Porras

**Review by uche-inyama — CHANGES_REQUESTED** [review]
## Changes Required

### Hi @NataJenkins :wave: 

After following the instructions on your readme file, I could not start the application locally on my machine. I got an error, I will leave a gif below for clarifications.

Peek 2021-04-27 22-42


Also, there is no **live demo link** on your readme file.

* [ ] I ran the application using vscode's live server, the form was not working and one of the fields **country code** you provided is not in the project requirements. Here is a gif.

error


* [ ] Also the feedback you got from your last review about using arrow functions hasn't been  
implemented.

#### Make these changes then submit another form for a code review.


### Happy coding :man_technologist: 

#### ``` Uchechukwu Henry Inyama ```

**Review by arslanbisharat — CHANGES_REQUESTED** [review]
# Changes Requested 💀 

Thank you for submitting the PR. 

As requested in previous [Code Reviews](https://github.com/NataJenkins/WeatherAppJs/pull/2#pullrequestreview-646396522) your project cannot be run. Please reach out to FT CRs for a 1:1 call to resolve your issue. 

Best, 
Arslan Bisharat
slack:arslanbisharat 🤙

**Review by arslanbisharat — CHANGES_REQUESTED** [review]
# Changes Requested 💀 

Thank you for submitting the PR. 

As requested in previous [Code Reviews](https://github.com/NataJenkins/WeatherAppJs/pull/2#pullrequestreview-646396522) your project cannot be run. Please reach out to FT CRs for a 1:1 call to resolve your issue. 

Please don't waste your Code Reviews.




Best, 
Arslan Bisharat.
slack:arslanbisharat 🤙

File: src/app/store.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    if (localStorage.getItem('countryCode') === null) {
+      this.countryCode = this.defaultCountry;
+    } else {
+      this.countryCode = localStorage.getItem('countryCode');
+    }
+    if (localStorage.getItem('units') === null) {
+      this.units = this.defaultUnits;
+    } else {
+      this.units = localStorage.getItem('units');
+    }
```
Comment (teekaytech) [inline src/app/store.js:26]: **[ OPTIONAL ]**

- You might want to change the long if-else statement in `src/app/store.js` to a switch/case statement.

File: src/app/ui.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    } else if (
+      weather.weather[0].description
+      === ('broken clouds' || 'shower rain' || 'rain' || 'thunderstorm' || 'mist')
+    ) {
+      background = 'rain';
+    } else if (weather.weather[0].description === 'clear sky') {
+      background = 'sun';
+    } else {
+      background = 'snow';
+    }
```
Comment (teekaytech) [inline src/app/ui.js:27]: **[ OPTIONAL ]**

- You can also change this to a switch/case statement.

**Review by teekaytech — CHANGES_REQUESTED** [review]
## Status: Changes Required :warning: :vertical_traffic_light: 

### Hello @NataJenkins :wave:

_I must commend your efforts so far in this project. Most of the features seem to be working._

__However, some changes are required before this project can be approved. Please check my comments below and fix them as appropriate.__

**[ REQUIRED CHANGES ]**

- [ ] As mentioned by @uche-inyama, you need to deploy your app and add the live demo link to the Readme.
- [ ] Also, you need to completely use the ES6 features. For instance, use `import` instead of `require`, as observed in `src/app/index.js`. 

* * *
**Please make the required changes and submit another review request when you are done.**

**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review unless it is requested otherwise.**

**Always post your inquiries and questions on this PR thread as a comment. You can mention me (@teekaytech) so I can see your comments and respond.**

* * *

**Taofeek Olalere
[Twitter](https://twitter.com/ola_lere) | [Github](https://github.com/teekaytech) | @taofeekolalere on Slack**
__Happy Coding :smiley:__

**Review by NtwaliHeritier — APPROVED** [review]
### **STATUS: APPROVED✔✔✔**

Great project👏👏
You can merge and move on👍✔

Happy coding💻💻💻

**Pull request merged**